src/partitioner/local/MetisPartitioner.cpp

```diff
@@ -385,6 +391,139 @@ void MetisPartitioner::createPartitionFiles(std::map<int, int> partMap) {
         this->sqlite.runUpdate(sqlStatement);
         dbLock.unlock();
     }
+
+    //Below code segment will create the composite central stores which uses for aggregation
+
+    if (nParts > 4) {
+        bool haveSwapped = true;
+
+        for (unsigned j = 1; haveSwapped && j < centralStoreSizeVector.size(); ++j) {
+            haveSwapped = false;
+
+            for (unsigned i = 0; i < centralStoreSizeVector.size() - j; ++i) {
+                if (centralStoreSizeVector[i] < centralStoreSizeVector[i+1]) {
+                    haveSwapped = true;
+
+                    size_t tempSize = centralStoreSizeVector[i];
+                    centralStoreSizeVector[i] = centralStoreSizeVector[i+1];
+                    centralStoreSizeVector[i+1] = tempSize;
+
+                    int tempPart = sortedPartVector[i];
+                    sortedPartVector[i] = sortedPartVector[i+1];
+                    sortedPartVector[i+1] = tempPart;
+                }
+            }
+        }
+
+        std::vector<size_t>::iterator centralStoreSizeVectorIterator;
+        std::map<int, std::vector<size_t>> centralStoreGroups;
+        std::map<int, std::vector<int>> partitionGroups;
+        std::vector<std::string> compositeGraphIdList;
+        int partitionLoop = 0;
+
+        for (centralStoreSizeVectorIterator = centralStoreSizeVector.begin(); centralStoreSizeVectorIterator != centralStoreSizeVector.end(); ++centralStoreSizeVectorIterator) {
+            size_t centralStoreSize = *centralStoreSizeVectorIterator;
+            std::vector<size_t> minSumGroup = centralStoreGroups[0];
+            std::vector<int> minPartitionGroup = partitionGroups[0];
+            std::vector<size_t>::iterator minSumGroupIterator;
+            size_t minGroupTotal = 0;
+            int minGroupIndex = 0;
+
+            for (minSumGroupIterator = minSumGroup.begin(); minSumGroupIterator != minSumGroup.end(); ++minSumGroupIterator) {
+                size_t size = *minSumGroupIterator;
+                minGroupTotal += size;
+            }
+
+            for (int loop = 0; loop < 4; loop++) {
```

Comment:
Why do we use 4 here?

Reply:
4 is the number of composite central stores

Reply:
Shall we use a constant here rather than using 4 then?

Reply:
Defined a constant and used it

Reply:
Done